lib/database/repo.ex

```diff
@@ -1,48 +1,55 @@
-defmodule BorsNG.Database.Repo do
+defmodule DynamicEctoRepoWrapper do
   @moduledoc """
-  An ecto data repository;
-  this process interacts with your persistent database.
-
-  Do not confuse this with a GitHub repo.
-  We call those `Project`s internally.
+  Exposes a macro to define Ecto.Repo functions dynamically.
   """
+  defmacro create_ecto_repo_callback(args, name) do
+    quote bind_quoted: [args: args, name: name] do
+      def unquote(name)(unquote_splicing(args)) do
+        repo = :persistent_term.get(:db_repo)
+        apply(repo, unquote(name), unquote(args))
```

Comment:
Not having much experience with macros, is `apply` really the only way to go here?